src/providers/slot/slot.ts

```diff
@@ -39,6 +39,47 @@ export class SlotProvider {
     });
   }
 
+  /**
+   * Extending the journal with empty days where there was no slots defined in the next 7 days
+   * @param slots Journal slots
+   * @returns Slots with additional empty days 
+   */
+  extendWithEmptyDays = (slots: {[k: string]: SlotItem[]}): {[k: string]: SlotItem[]} => {
+    const emptyDays = {
+      [moment().add(0, 'day').format('YYYY-MM-DD')]: [],
+      [moment().add(1, 'day').format('YYYY-MM-DD')]: [],
+      [moment().add(2, 'day').format('YYYY-MM-DD')]: [],
+      [moment().add(3, 'day').format('YYYY-MM-DD')]: [],
+      [moment().add(4, 'day').format('YYYY-MM-DD')]: [],
+      [moment().add(5, 'day').format('YYYY-MM-DD')]: [],
+      [moment().add(6, 'day').format('YYYY-MM-DD')]: [],
+    };
+
+    return {
+      ...emptyDays,
+      ...slots,
+    }
+  }
+
+  /**
+   * Slice the journal slots and get the slots only for the relevant days
+   * | From regular working weekday we can see the next working weekday
+   * | From Friday we can navigate through weekend till Monday
+   * @param slots Journal slots
+   * @returns Only the relevant slots
+   */
+  getRelevantSlots = (slots: {[k: string]: SlotItem[]}): {[k: string]: SlotItem[]} => {
```

Comment:
do we have unit tests for these new methods?

Reply:
We have one test for `extendWithEmptyDays` function. But we don't have any for the `getRelevantSlots` and the reason for this is that function is heavily dependent on `moment()`. So by this we are highlighting the fact that we need to abstract the way that we are working with date and time and can't just have our code being polluted by `moment.js`.

Reply:
We are going to build an abstraction layer around `moment.js` so that we can control the features better. And that time will write the tests for all the different cases what we left out while we have been depending on the framework.